src/components/misc/loclist/LocationList.jsx

```diff
@@ -38,7 +38,7 @@ export default class LocationList extends React.Component {
                         return (
                             <LocationListItem location={ loc }
                                 key={ loc.data.id }
-                                onSelect={ this.onSelect.bind(this, loc) }/>
+                                onSelect={ this.onSelect.bind(this, loc.data) }/>
```

Comment:
Please, if it's possible, undo this action, instead changing `components/sections/maps/LocationsPane` so that the `data` attribute is retrieved from the item object there.

That way, we can stick to a single pattern where lists dispatch the entire item (introduced in #294).